File: backend/infrastructure/repository/model/article.go
package model

import (
	"home/entity"
	"log"

	"gorm.io/gorm"
)

type Article struct {
	ID        string
	Title     string
	Content   string
	Tags      []*Tag `gorm:"many2many:article_tags"`
	Viewers   uint
	Favorites uint
	gorm.Model
}

func (m Article) ToEntity() *entity.Article {
	tags := []*entity.Tag{}
	if m.Tags != nil {
		for _, tag := range m.Tags {
			tags = append(tags, tag.ToEntity())
		}
	}
	log.Print(m.Tags)

	return &entity.Article{
		ID:        m.ID,
		Title:     m.Title,
		Content:   m.Content,
		Tags:      tags,
		Viewers:   m.Viewers,
		Favorites: m.Favorites,
	}
}

File: frontend/src/model/Article.ts
import { z } from "zod";

import { Tag } from "./Tag";

export const Article = z.object({
  id: z.string(),
  title: z.string(),
  content: z.string(),
  tags: z.array(z.nullable(Tag)),
  viewers: z.number().int().min(0),
  favorites: z.number().int().min(0),
});

export type Article = z.infer<typeof Article>;
